TelepresenceBot/server-unit/core/disconnector.js

```diff
@@ -0,0 +1,28 @@
+function Disconnector(rooms, connectedClients) {
+    return {
+        disconnectRoom: function(roomName) {
+            var room = rooms[roomName];
+
+            if(!room) {
+                return false;
+            }
+
+            var clientsInRoom = room.sockets;
+
+            for(var client in clientsInRoom) {
+                var connectedClient = connectedClients[client];
+
+                if(!connectedClient){
+                    return false;
+                }
+
+                connectedClient.disconnect();
+            }
+            return true;
+        }
+    };
+}
+
+module.exports = function(rooms, connectedClients) {
+    return new Disconnector(rooms, connectedClients);
```

Comment:
since you aren't retaining any state you don't _reallly_ need to new these up.

If you want to keep and treat them like classes, you should probably use classes 👻 

```

module.exports = class Disconnector {

    disconnectRoom(roomName) {
        this._anInternalMethod()
    }

    _anInternalMethod() {
    }

}
```

and then the usage convention is

```
const Disconnector = require('./diconnector')
cont disconnector = new Disconnector()
```